internal/builtinfunctions/functions_test.go

```diff
@@ -0,0 +1,471 @@
+		false,
+		true,
+	},
+	"string-to-bool-3": {
+		"stringToBool",
+		[]any{"TRUE"},
+		false,
+		true,
+	},
+	"string-to-bool-4": {
+		"stringToBool",
+		[]any{"t"},
+		false,
+		true,
+	},
+	"string-to-bool-5": {
+		"stringToBool",
+		[]any{"T"},
+		false,
+		true,
+	},
+	"string-to-bool-6": {
+		"stringToBool",
+		[]any{"1"},
+		false,
+		true,
+	},
+	"string-to-bool-7": {
+		"stringToBool",
+		[]any{"false"},
+		false,
+		false,
+	},
+	"string-to-bool-8": {
+		"stringToBool",
+		[]any{"False"},
+		false,
+		false,
+	},
+	"string-to-bool-9": {
+		"stringToBool",
+		[]any{"FALSE"},
+		false,
+		false,
+	},
+	"string-to-bool-10": {
+		"stringToBool",
+		[]any{"f"},
+		false,
+		false,
+	},
+	"string-to-bool-11": {
+		"stringToBool",
+		[]any{"F"},
+		false,
+		false,
+	},
+	"string-to-bool-12": {
+		"stringToBool",
+		[]any{"0"},
+		false,
+		false,
+	},
+	"string-to-bool-error-1": {
+		"stringToBool",
+		[]any{""},
+		true,
+		nil,
+	},
+	"string-to-bool-error-2": {
+		"stringToBool",
+		[]any{"abc"},
+		true,
+		nil,
+	},
+	"ceil-1": {
+		"ceil",
+		[]any{0.0},
+		false,
+		0.0,
+	},
+	"ceil-2": {
+		"ceil",
+		[]any{0.1},
+		false,
+		1.0,
+	},
+	"ceil-3": {
+		"ceil",
+		[]any{-0.1},
+		false,
+		0.0,
+	},
+	"floor-1": {
+		"floor",
+		[]any{0.0},
+		false,
+		0.0,
+	},
+	"floor-2": {
+		"floor",
+		[]any{1.1},
+		false,
+		1.0,
+	},
+	"floor-3": {
+		"floor",
+		[]any{2.9999},
+		false,
+		2.0,
+	},
+	"floor-4": {
+		"floor",
+		[]any{-0.1},
+		false,
+		-1.0,
+	},
+	"round-1": {
+		"round",
+		[]any{0.0},
+		false,
+		0.0,
+	},
+	"round-2": {
+		"round",
+		[]any{0.1},
+		false,
+		0.0,
+	},
+	"round-3": {
+		"round",
+		[]any{0.4},
+		false,
+		0.0,
+	},
+	"round-4": {
+		"round",
+		[]any{0.5},
+		false,
+		1.0,
+	},
+	"round-5": {
+		"round",
+		[]any{0.9},
+		false,
+		1.0,
+	},
+	"round-6": {
+		"round",
+		[]any{-0.9},
+		false,
+		-1.0,
+	},
+	"round-7": {
+		"round",
+		[]any{-0.5},
+		false,
+		-1.0,
+	},
+	"round-8": {
+		"round",
+		[]any{-0.01},
+		false,
+		0.0,
+	},
+	"abs-1": {
+		"abs",
+		[]any{0.0},
+		false,
+		0.0,
+	},
+	"abs-2": {
+		"abs",
+		[]any{-1.0},
+		false,
+		1.0,
+	},
+	"abs-3": {
+		"abs",
+		[]any{1.0},
+		false,
+		1.0,
+	},
+	"to-lower-1": {
+		"toLower",
+		[]any{""},
+		false,
+		"",
+	},
+	"to-lower-2": {
+		"toLower",
+		[]any{"abc"},
+		false,
+		"abc",
+	},
+	"to-lower-3": {
+		"toLower",
+		[]any{"ABC"},
+		false,
+		"abc",
+	},
+	"to-upper-1": {
+		"toUpper",
+		[]any{""},
+		false,
+		"",
+	},
+	"to-upper-2": {
+		"toUpper",
+		[]any{"abc"},
+		false,
+		"ABC",
+	},
+	"to-upper-3": {
+		"toUpper",
+		[]any{"ABC"},
+		false,
+		"ABC",
+	},
+	"split-string-1": {
+		"splitString",
+		[]any{"", ","},
+		false,
+		make([]string, 1), // The size appears to matter with DeepEquals
+	},
+	"split-string-2": {
+		"splitString",
+		[]any{"abc", ","},
+		false,
+		[]string{"abc"},
+	},
+	"split-string-3": {
+		"splitString",
+		[]any{"a,b", ","},
+		false,
+		[]string{"a", "b"},
+	},
+	"split-string-4": {
```

Comment:
How about ",1,2" and "1,2,", or even "1,,2"? I know how the Python splitter works with those, I'm not sure about Go (sure, I could look it up), and demonstrating with test cases could be handy...